### ISSUE TITLE
why you use sampled_label for the combined model instead of labels and didn't set "self.discriminator.trainable = True" for training the discriminator?

### ISSUE THREAD

**Issue by DushyantChauhan** [eriklindernoren/Keras-GAN#182]
Hello Erik,
(1) I just want to understand why you use sampled_label for the combined model instead of labels?
g_loss = self.combined.train_on_batch([noise, sampled_labels], valid).

(2) you set "self.discriminator.trainable = False" to train the generator but for training the discriminator why did you not set "self.discriminator.trainable = Ture"?

Thank you.

**Comment by TheGuywithTheHat** [comment]
regarding (2), see: https://github.com/eriklindernoren/Keras-GAN/issues/22#issuecomment-379553290

Essentially, it is important that `trainable = false` is set between `discriminator.compile()` and `combined.compile()`.

**Comment by hiteshnitetc** [comment]
For which gan you are talking about.
